### DISCUSSION TITLE
/thread/thread-state-and-method.html

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#39] [Announcements]
# /thread/thread-state-and-method.html

Java线程的6种状态及切换(透彻讲解)

https://tobebetterjavaer.com/thread/thread-state-and-method.html

**Comment by xiaoacaia** [comment]
至于b线程的状态，有可能打印RUNNABLE（尚未进入同步方法），也有可能打印TIMED_WAITING（进入了同步方法）

请问括号内的内容是不是标反了 ？

**Comment by yeahQing** [comment]
如果CPU执行效率高一点，还没等两个线程真正开始争夺锁，就已经打印此时两个线程的状态（RUNNABLE）了。怎么会出现两个线程都再RUNNABLE状态呢?

**Reply by yeahQing** [reply]
明白了，a和b线程调用start就会是RUNNABLE状态，不过这个状态的时间很短，之后马上就会争抢锁，所以如果CPU执行效率高，就可以让启动线程的速度变快，这个时候main线程就有机会输出a，b线程都在RUNNABLE状态。
